torch_xla/experimental/tpu.py

```diff
@@ -127,8 +127,7 @@ def version() -> int:
   except requests.HTTPError as e:
     raise EnvironmentError('Failed to get TPU metadata') from e
 
-  match = re.match(r'^v(\d)([A-Za-z]?){7}-(\d+)$', env[xenv.ACCELERATOR_TYPE])
-  return int(match.groups()[0])
+  return int((env[xenv.ACCELERATOR_TYPE])[1])
```

Comment:
IMO the regex here is clearer. Do you have a reason to change it? The existing one works in all of the test cases:

```
>>> re.match(r'^v(\d)([A-Za-z]?){7}-(\d+)$', 'v5abcdefg-4').groups()[0]
'5'
>>> re.match(r'^v(\d)([A-Za-z]?){7}-(\d+)$', 'v5abcd-4').groups()[0]
'5'
>>> re.match(r'^v(\d)([A-Za-z]?){7}-(\d+)$', 'v5-4').groups()[0]
'5'
```

Reply:
Thanks, updated

Reply:
Thanks, Will @will-cromar , after the modification based on your advice, current code has passed the CI tests too, do we want to mark this request change as finished and submit this PR?